commit 5bdcb3ea9afdefd73aa1ee5a4f0967ed509a29a5
Author: bertin <macportsuser@macports.org>
Date:   Thu May 22 16:10:05 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-macports.tcl-environment-option.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index 091179ea9..fccb9053a 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -132,6 +132,59 @@ proc macports::version {} {
     return ${macports::autoconf::macports_version}
 }
 
+##
+# Return the port environment variables.
+proc macports::environment {} {
+    set mpenv {}
+    if {![macports::ui_isset ports_quiet]} {
+        lappend mpenv "$macports::ui_prefix port_cmd_version:  ${macports::autoconf::macports_version}"
+    } else {
+        lappend mpenv "port_cmd_version  ${macports::autoconf::macports_version}"
+    }
+    if {[info exists macports::bootstrap_options]} {
+        foreach bootstrap_option [lsort $macports::bootstrap_options] {
+            if {[info exists macports::$bootstrap_option]} {
+                if {![macports::ui_isset ports_quiet]} {
+                    lappend mpenv "$macports::ui_prefix bootstrap_options: $bootstrap_option: [set macports::$bootstrap_option]"
+                } else {
+                    lappend mpenv "bootstrap_options $bootstrap_option [set macports::$bootstrap_option]"
+                }
+            }
+        }
+    }
+    if {[info exists macports::sources]} {
+        foreach source $macports::sources {
+            if {![macports::ui_isset ports_quiet]} {
+                lappend mpenv "$macports::ui_prefix port_tree_sources: $source"
+            } else {
+                lappend mpenv "port_tree_sources $source"
+            }
+        }
+    }
+    if {[info exists macports::global_variations]} {
+        set gvl {}
+        foreach {variation mode} [array get macports::global_variations] {
+            lappend gvl $mode$variation
+        }
+        if {![macports::ui_isset ports_quiet]} {
+            lappend mpenv "$macports::ui_prefix global_variations: [join $gvl { }]"
+        } else {
+            lappend mpenv "global_variations [join $gvl { }]"
+        }
+    }
+    set shenv {}
+    foreach keyval [exec /usr/bin/printenv] {
+        lappend shenv $keyval
+    }
+    if {![macports::ui_isset ports_quiet]} {
+        lappend mpenv "$macports::ui_prefix shell environment: [join $shenv { }]"
+    } else {
+        lappend mpenv "shell environment [join $shenv { }]"
+    }
+    ui_msg "[join $mpenv \n]"
+    return 0
+}
+
 # Provided UI instantiations
 # For standard messages, the following priorities are defined
 #     debug, info, msg, warn, error
@@ -1143,6 +1196,9 @@ proc mportinit {{up_ui_options {}} {up_options {}} {up_variations {}}} {
                     } else {
                         set $option $val
                     }
+                    if {[info exists macports::ui_options(ports_env)]} {
+                        ui_msg "$macports::ui_prefix $file: ${option}: [set $option]"
+                    }
                 }
                 set continuation $next_continuation
             }
@@ -1159,6 +1215,9 @@ proc mportinit {{up_ui_options {}} {up_options {}} {up_variations {}}} {
                 if {$option in $user_options} {
                     global macports::$option
                     set $option $val
+                    if {[info exists macports::ui_options(ports_env)]} {
+                        ui_msg "$macports::ui_prefix user: ${option}: [set $option]"
+                    }
                 }
             }
         }
@@ -1209,6 +1268,9 @@ Please edit sources.conf and change '$url' to '[string range $url 0 26]macports/
                     }
                 }
                 lappend sources [concat [list $url] $flags]
+                if {[info exists macports::ui_options(ports_env)]} {
+                    ui_msg "$macports::ui_prefix $sources_conf: [lindex $sources end]"
+                }
             } else {
                 ui_warn "$sources_conf specifies invalid source '$line', ignored."
             }
